DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs

```diff
@@ -1,222 +1,250 @@
+                                AND P.RemovedDate IS NULL
+                                AND p.LoginCount = 0)
+                        AND ({numberOfGroupsWhereDelegateIsEnrolledOnThisCourse}) = 1",
+                new { groupId, delegateId, removedDate }
+            );
+        }
+
+        public int? GetRelatedProgressIdForGroupDelegate(int groupId, int delegateId)
+        {
+            return connection.Query<int?>(
+                @"SELECT ProgressID
+                    FROM Progress AS P
+                    INNER JOIN GroupCustomisations AS GC ON P.CustomisationID = GC.CustomisationID
+                    WHERE p.Completed IS NULL
+                        AND p.EnrollmentMethodID = 3
+                        AND GC.GroupID = @groupId
+                        AND p.CandidateID = @delegateId
+                        AND P.RemovedDate IS NULL",
+                new { groupId, delegateId }
+            ).FirstOrDefault();
+        }
+
+        public void DeleteGroupDelegatesRecordForDelegate(int groupId, int delegateId)
+        {
+            connection.Execute(
+                @"DELETE FROM GroupDelegates
+                    WHERE GroupID = @groupId
+                      AND DelegateID = @delegateId",
+                new { groupId, delegateId }
+            );
+        }
+
+        public int AddDelegateGroup(GroupDetails groupDetails)
+        {
+            return connection.QuerySingle<int>(
+                @"INSERT INTO Groups (CentreID, GroupLabel, GroupDescription, LinkedToField, SyncFieldChanges, AddNewRegistrants, PopulateExisting, CreatedDate, CreatedByAdminUserID)
+                        OUTPUT inserted.GroupID
+                        VALUES (@CentreId, @GroupLabel, @GroupDescription, @LinkedToField, @SyncFieldChanges, @AddNewRegistrants, @PopulateExisting, @CreatedDate, @AdminUserId)",
+                groupDetails
+            );
+        }
+
+        public void AddDelegateToGroup(int delegateId, int groupId, DateTime addedDate, int addedByFieldLink)
+        {
+            connection.Execute(
+                @"INSERT INTO GroupDelegates (GroupID, DelegateID, AddedDate, AddedByFieldLink)
+                    VALUES (
+                        @groupId, 
+                        @delegateId, 
+                        @addedDate, 
+                        @addedByFieldLink)",
+                new { groupId, delegateId, addedDate, addedByFieldLink }
+            );
+        }
+
+        public Group? GetGroup(int groupId, int centreId)
+        {
+            return connection.Query<Group>(
+                @"SELECT
+                        GroupLabel,
+                        GroupDescription
+                    FROM Groups
+                    WHERE GroupID = @groupId AND CentreId = @centreId",
+                new { groupId, centreId }
+            ).SingleOrDefault();
+        }
+
+        public bool UpdateGroupDescription(int groupId, int centreId, string groupDescription)
```

Comment:
The name of this isn't clear enough as to what is happening, we need to name it so that it is obvious that this is expected to sometimes fail and not actually update the database. Maybe Try..., or UpdateGroupDescriptionAndReturnIfSuccessful or something. The name of the unit tests then need updating to match.

Reply:
I updated method name to IsGroupDescriptionUpdated

Reply:
I don't think IsGroupDescriptionUpdated is the correct name here, that suggests the method is checking if the group name has been updated or not rather than it is going to do an update. TryUpdateGroupDescription is perhaps closer to what we need, following a similar convention to TryParse.

Reply:
Method name is udpated to TryUpdateGroupDescription